genie-agent/src/main/java/com/netflix/genie/agent/execution/services/impl/JobSetupServiceImpl.java

```diff
@@ -710,6 +710,12 @@ String composeScript(final JobSetupServiceProperties jobSetupProperties) {
                 .append("# Launch the command")
                 .append(NEWLINE)
                 .append(this.commandLine).append(" <&0 &").append(NEWLINE)
+                // Spawn a new child process to watch if parent process dies, if so kill command forcefully
+                .append("pid=$!").append(NEWLINE)
+                .append("ppid=$$").append(NEWLINE)
+                .append("{ while kill -0 $ppid &> /dev/null; do sleep 30; done; "
```

Comment:
Do you think 30s is too long of a gap? Should it possibly be shorter? We also could make this a property at some point